### ISSUE TITLE
Single-node CoreOS/kubernetes getting-started-guides addition

### ISSUE THREAD

**Issue by commixon** [kubernetes/kubernetes#16847]
My proposed addition is a cloud-config file for a single node CoreOS/kubernetes setup. This cloud-config file works for every provider that supports cloud-init files (eg. vagrant, openstack, ec2 etc...) and it is also configured to setup the dns-addon by default. It is meant for development purposes only, but i think it would be a nice addition in the getting-started-guides section.
What would you think, is this a valid reason for such a PR?

**Comment by dchen1107** [comment]
@yifan-gu is working on something similar to this.

**Comment by commixon** [comment]
Oh, so should I just close the Issue then and not proceed to a PR?

**Comment by yifan-gu** [comment]
@commixon I am working a multinode setup for coreos/gce particularly at this point. Haven't thought about how you can have a generic cloudconfig for every platform?

**Comment by commixon** [comment]
Well my proposal is not for multi-node setup, especially with such differences between providers. However I use a cloud-init file to launch a single, all-in-one kubernetes on CoreOS, that works both for local environments (e.g. vagrant) and others as well (I have tested and use it with openstack, ec2, digitalocean and azure). There is no need for complex scenarios, just use the $private_ipv4 and $public_ipv4 to configure everything to listen on localhost and the apiserver on <public>:8080.

Since this all is aimed for development and development only, I think it would be a nice addition to the getting started guides. I have also added a OneShot unit, that waits for kubelet to come up and adds (through curl and json) the sky-dns rc and service as well. Again, this is just for development purposes but it has helped my team a lot, since we need quick and dirty development environments, both through vagrant or public providers.
